parsons/empower/empower.py

```diff
@@ -0,0 +1,261 @@
+from parsons.utilities.api_connector import APIConnector
+from parsons.utilities import check_env
+from parsons.etl import Table
+import logging
+from datetime import datetime
+
+logger = logging.getLogger(__name__)
+
+EMPOWER_API_ENDPOINT = "https://api.getempower.com/v1/export"
+
+
+class Empower(object):
+    """
+    Instantiate class.
+
+        `Args:`
+            api_key: str
+                The Empower provided API key.The Empower provided Client UUID. Not
+                required if ``EMPOWER_API_KEY`` env variable set.
+            empower_uri: str
+                The URI to access the Empower API. The default is currently set to
+                https://api.getempower.com/v1/export. You can set an ``EMPOWER_URI`` env
+                variable or use this URI parameter if a different endpoint is necessary.
+            cache: boolean
+                The Empower API returns all account data after each call. Setting cache
+                to ``True`` stores the blob and then extracts Parsons tables for each method.
+                Setting cache to ``False`` will download all account data for each method call.
+    """
+
+    def __init__(self, api_key=None, empower_uri=None, cache=True):
+        self.api_key = check_env.check("EMPOWER_API_KEY", api_key)
+        self.empower_uri = (
+            check_env.check("EMPOWER_URI", empower_uri, optional=True) or EMPOWER_API_ENDPOINT
+        )
+        self.headers = {"accept": "application/json", "secret-token": self.api_key}
+        self.client = APIConnector(
+            self.empower_uri,
+            headers=self.headers,
+        )
+        self.data = None
+        self.data = self._get_data(cache)
+
+    def _get_data(self, cache):
+        """
+        Gets fresh data from Empower API based on cache setting.
+        """
+
+        if not cache or self.data is None:
+            r = self.client.get_request(self.empower_uri)
+            logger.info("Empower data downloaded.")
+            return r
+
+        else:
+            return self.data
+
+    def _unix_convert(self, ts):
+        """
+        Converts UNIX timestamps to readable timestamps.
+        """
+
+        ts = datetime.utcfromtimestamp(int(ts) / 1000)
+        ts = ts.strftime("%Y-%m-%d %H:%M:%S UTC")
+        return ts
+
+    def _empty_obj(self, obj_name):
+        """
+        Determine if a dict object is empty.
+        """
+
+        if len(self.data[obj_name]) == 0:
+            return True
+        else:
+            return False
+
+    def get_profiles(self):
+        """
+        Get Empower profiles.
+
+        `Returns:`
+            Parsons Table
+                See :ref:`parsons-table` for output options.
+        """
+
+        tbl = Table(self.data["profiles"])
+        for col in ["createdMts", "lastUsedEmpowerMts", "updatedMts"]:
+            tbl.convert_column(col, lambda x: self._unix_convert(x))
+        tbl.remove_column("activeCtaIds")  # Get as a method via get_profiles_active_ctas
```

Comment:
What is this line doing? I don't understand the code comment

Reply:
I think the comment means that if you want this column, you would get it using get_profiles_active_ctas